defaults/main.yml

```diff
@@ -4,9 +4,21 @@ sansible_gocd_agent_auto_register_environments: ~
 sansible_gocd_agent_auto_register_hostname: ~
 sansible_gocd_agent_auto_register_key: ~
 sansible_gocd_agent_auto_register_resources: ~
+
+# DEPRECARTED sansible_gocd_agent_aws_gocd_server_lookup_filter
```

Comment:
I would remove the old way completely and bump the major version, no need to carry over tech debt.

Reply:
👍 will do that

Reply:
Yes, please.  I left the old way (with the regexp hacks) specifically to not raise the major version, but if we do this, then let's do this properly :-)